### PR TITLE
feat: add alias for new and existing index

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:

* Add alias for new and existing index on update/create
* Use alias for update/create
* Tested for 
  * Alias is created for existing index upon update/create
  * New index is created with alias 
  * Create, update and delete patient using index alias 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/ddbToEs/ddbToEsHelper.ts [hunk 1]
```diff
@@ -43,8 +43,10 @@ export default class DdbToEsHelper {
     }
 
     async createIndexIfNotExist(indexName: string) {
+        logger.info('entering create index function');
```
Comment (carvantes) [inline src/ddbToEs/ddbToEsHelper.ts]: change to debug level or remove.

File: src/ddbToEs/ddbToEsHelper.ts [hunk 2]
```diff
@@ -70,12 +72,27 @@ export default class DdbToEsHelper {
                                 },
                             },
                         },
+                        [`${indexName}-alias`]: {},
                     },
                 };
                 await this.ElasticSearch.indices.create(params);
+            } else {
```
Comment (carvantes) [inline src/ddbToEs/ddbToEsHelper.ts:79]: I think that the if conditions are slightly off.
If the index does not exists, then we should create the index AND the alias. (right now the alias is only created if the index existed before this method call)

something like:
```js
if (aliasExists){
	return
}

if(!indexExists){
	createIndex()
}

createAlias()
```

File: src/ddbToEs/ddbToEsHelper.test.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import { Client } from '@elastic/elasticsearch';
+
+import Mock from '@elastic/elasticsearch-mock';
```
Comment (carvantes) [inline src/ddbToEs/ddbToEsHelper.test.ts:3]: 👍

**Review by carvantes — CHANGES_REQUESTED** [review]
Looks good so far, but I left a comment that needs to be addressed before merging.

Also, we need to make a change to write to aliases instead of indexes at: https://github.com/awslabs/fhir-works-on-aws-persistence-ddb/blob/0da8a2866346ad54929c0efc7fdbbb61d957c1eb/src/ddbToEs/ddbToEsHelper.ts#L127

Otherwise the aliases are meaningless.

File: src/ddbToEs/ddbToEsHelper.ts (hunk 2 above)
Comment (Bingjiling) [reply src/ddbToEs/ddbToEsHelper.ts:79]: The create index and alias scenario is handled by line 75, where the alias is created as part of the create index call.

Comment (carvantes) [reply src/ddbToEs/ddbToEsHelper.ts:79]: My bad.

Is line 75 correct? Shouldn't it be inside a "aliases" object?
i.e.
```js
"aliases": {
    [`${indexName}-alias`]: {},
  }
```

https://www.elastic.co/guide/en/elasticsearch/reference/7.x/indices-create-index.html#create-index-aliases

Comment (Bingjiling) [reply src/ddbToEs/ddbToEsHelper.ts:79]: Good catch!

**Review by carvantes — APPROVED** [review]

**Pull request merged**